editor/filesystem_dock.cpp

```diff
@@ -437,39 +434,15 @@ void FileSystemDock::_update_files(bool p_keep_selection) {
 		files->set_max_text_lines(2);
 		files->set_fixed_icon_size(Size2(thumbnail_size, thumbnail_size));
 
-		if (!has_icon("ResizedFolder", "EditorIcons")) {
-			Ref<ImageTexture> folder = get_icon("FolderBig", "EditorIcons");
-			Ref<Image> img = folder->get_data();
-			img = img->duplicate();
-			img->resize(thumbnail_size, thumbnail_size);
-			Ref<ImageTexture> resized_folder = Ref<ImageTexture>(memnew(ImageTexture));
-			resized_folder->create_from_image(img, 0);
-			Theme::get_default()->set_icon("ResizedFolder", "EditorIcons", resized_folder);
-		}
-
-		folder_thumbnail = get_icon("ResizedFolder", "EditorIcons");
-
-		if (!has_icon("ResizedFile", "EditorIcons")) {
-			Ref<ImageTexture> file = get_icon("FileBig", "EditorIcons");
-			Ref<Image> img = file->get_data();
-			img->resize(thumbnail_size, thumbnail_size);
-			Ref<ImageTexture> resized_file = Ref<ImageTexture>(memnew(ImageTexture));
-			resized_file->create_from_image(img, 0);
-			Theme::get_default()->set_icon("ResizedFile", "EditorIcons", resized_file);
-		}
-
-		if (!has_icon("ResizedFileBroken", "EditorIcons")) {
-			Ref<ImageTexture> file = get_icon("FileBigBroken", "EditorIcons");
-			Ref<Image> img = file->get_data();
-			img->resize(thumbnail_size, thumbnail_size);
-			Ref<ImageTexture> resized_file = Ref<ImageTexture>(memnew(ImageTexture));
-			resized_file->create_from_image(img, 0);
-			Theme::get_default()->set_icon("ResizedFileBroken", "EditorIcons", resized_file);
+		if (thumbnail_size < 64) {
+			folder_thumbnail = get_icon("FolderMediumThumb", "EditorIcons");
+			file_thumbnail = get_icon("FileMediumThumb", "EditorIcons");
+			file_thumbnail_broken = get_icon("FileDeadMediumThumb", "EditorIcons");
+		} else {
+			folder_thumbnail = get_icon("FolderBigThumb", "EditorIcons");
+			file_thumbnail = get_icon("FileBigThumb", "EditorIcons");
+			file_thumbnail_broken = get_icon("FileDeadBigThumb", "EditorIcons");
 		}
-
-		file_thumbnail = get_icon("ResizedFile", "EditorIcons");
-
-		file_thumbnail_broken = get_icon("ResizedFileBroken", "EditorIcons");
```

Comment:
just so I'm understanding. You do the resizing inside the theme generation. as a consequence you can just load the icon and it will be the correct size so no need to buffer it in the default theme?

Reply:
@toger5 yeah.